comps/agent/langchain/src/utils.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+
+from .config import env_config
+
+
+def format_date(date):
+    # input m/dd/yyyy hr:min
+    # output yyyy-mm-dd
+    date = date.split(" ")[0]  # remove hr:min
+    # print(date)
+    try:
+        date = date.split("/")  # list
+        # print(date)
+        year = date[2]
+        month = date[0]
+        if len(month) == 1:
+            month = "0" + month
+        day = date[1]
+        return f"{year}-{month}-{day}"
+    except:
+        return date
+
+
+def setup_hf_tgi_client(args):
+    from langchain_huggingface import HuggingFaceEndpoint
+
+    generation_params = {
```

Comment:
@lvliang-intel should we set the generation config here or a specific config file

Reply:
I see gateway is also using some pre-defined parameters, if some of the parameters worth for exposing to users, we can update that later?